components/brave_wallet_ui/components/extension/sign-panel/sign_in_with_ethereum_error.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useDispatch } from 'react-redux'
+import Button from '@brave/leo/react/button'
+
+// Types
+import {
+  BraveWallet
+} from '../../../constants/types'
+
+// Utils
+import {
+  reduceAddress
+} from '../../../utils/reduce-address'
+import { getLocale } from '../../../../common/locale'
+
+// Selectors
+import {
+  useUnsafePanelSelector
+} from '../../../common/hooks/use-safe-selector'
+import {
+  PanelSelectors
+} from '../../../panel/selectors'
+
+// Components
+import CreateSiteOrigin from '../../shared/create-site-origin/index'
+
+import {
+  StyledWrapper,
+  Title,
+  ErrorBox,
+  ErrorDescriptionText,
+  WarningIcon,
+  OriginErrorText,
+  LaunchButton,
+  LaunchIcon
+} from './sign_in_with_ethereum.style'
+import {
+  Row,
+  Column,
+  VerticalDivider,
+  VerticalSpace
+} from '../../shared/style'
+import { PanelActions } from '../../../panel/actions'
+
+export const SignInWithEthereumError = () => {
+  // Redux
+  const dispatch = useDispatch()
+
+  // Selectors
+  const signMessageErrorData =
+    useUnsafePanelSelector(PanelSelectors.signMessageErrorData)
+
+  // Memos
+  const errorData = React.useMemo(() => {
+    if (signMessageErrorData.length === 0) {
+      return undefined
+    }
+    return signMessageErrorData[0]
+  }, [signMessageErrorData])
+
+  const errorMessage = React.useMemo(() => {
```

Comment:
Can we redfine this so it can be defined outside of the component?

Reply:
Changed to `Computed`